qsimcirq_tests/qsimcirq_test.py

```diff
@@ -1376,3 +1376,123 @@ def test_cirq_qsim_global_shift():
     assert cirq.linalg.allclose_up_to_global_phase(
         qsim_result.state_vector(), cirq_result.state_vector()
     )
+
+
+@pytest.mark.parametrize("mode", ["noiseless", "noisy"])
+def test_cirq_qsim_circuit_memoization(mode: str):
+    """Verifies the correctness of simulator functions when
+    circuit_memoization_size is set."""
+    execution_repetitions = 3
+    qsim_sim = qsimcirq.QSimSimulator(circuit_memoization_size=4)
+    cirq_sim = cirq.Simulator()
+
+    # Pick qubits.
+    a, b, c, d = [
+        cirq.GridQubit(0, 0),
+        cirq.GridQubit(0, 1),
+        cirq.GridQubit(1, 1),
+        cirq.GridQubit(1, 0),
+    ]
+
+    # compute_amplitudes()
```

Comment:
Could you split these (`compute_amplitudes`, `simulate`, etc.) into separate test methods?

Reply:
done :)